.github/workflows/junit-test-reporter.yml

```
@@ -0,0 +1,23 @@
+---
+## Workflow to process the JUnit test results and add a report to the checks.
+name: Test Report
+on:
+  workflow_run:
+    workflows:
+      - pull-request
+    types:
+      - completed
+
+permissions:
+  contents: read
+
+jobs:
+  report:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dorny/test-reporter@c9b3d0e2bd2a4e96aaf424dbaa31c46b42318226
```

Use https://github.com/elastic/apm-pipeline-library/tree/main/.github/actions/test-report

```suggestion
      - uses: elastic/apm-pipeline-library/.github/actions/test-report@gh_actions
```